client/state/data-layer/wpcom/sites/hosting/sftp-user.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import {
+	HOSTING_SFTP_USER_REQUEST,
+	HOSTING_SFTP_USER_CREATE,
+	HOSTING_SFTP_PASSWORD_RESET,
+} from 'state/action-types';
+import { errorNotice } from 'state/notices/actions';
+import { translate } from 'i18n-calypso';
+import { updateAtomicSFTPUser } from 'state/hosting/actions.js';
+
+const fromApi = response => {
+	if ( ! response || ! response.username ) {
+		throw new Error( 'Retrieving or updating the sftp user was unsuccessful', response );
+	}
+
+	return response.password
+		? { username: response.username, password: response.password }
+		: { username: response.username };
+};
+
+const requestAtomicSFTPUser = action => {
+	return http(
+		{
+			method: 'GET',
+			path: `/sites/${ action.siteId }/hosting/ssh-user`,
+			apiNamespace: 'wpcom/v2',
+		},
+		action
+	);
+};
+
+const createAtomicSFTPUser = action => {
+	return http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/hosting/ssh-user`,
+			apiNamespace: 'wpcom/v2',
+			body: {},
+		},
+		action
+	);
+};
+
+const resetAtomicSFTPPassword = action => {
+	return http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/hosting/ssh-user/reset-password`,
+			apiNamespace: 'wpcom/v2',
+			body: {},
+		},
+		action
+	);
+};
+
+const receiveAtomicSFTPUserSuccess = ( action, sftpUser ) =>
```

Comment:
Given the API response is rather simple, I think we can parse it here and drop the `fromApi` usage:

```js
const receiveAtomicSftpUserSuccess = ( 
	{ siteId, userId }, 
	{ username, password: undefined }
) => updateAtomicSftpUser( siteId, userId, { username, password } );
```

Note that we don't do the `! response || ! response.username` check here, but I think we can assume the API will return always a response and a username. If that's not the case, then the `onError` callback will be triggered.

Reply:
Yes, my approach was probably overkill  - have simplified as suggested.